File: member-service/src/main/java/com/hedvig/memberservice/web/dto/CashbackOption.java
package com.hedvig.memberservice.web.dto;

import lombok.Value;
import lombok.experimental.Wither;

import java.util.UUID;
/*        {
        "id": "someid",
        "title": "Rädda Barnen",
        "description": "Lorem ipsum dolor sit amet...",
        "selected": false,
        "charity": true,
        "imageUrl": "https://unsplash.it/400/200"
        },
        {
        "id": "someotherid",
        "title": "Mitt konto",
        "description": "Lorem ipsum dolor sit amet...",
        "selected": true,
        "charity": false,
        "imageUrl": "https://unsplash.it/400/200"
        }
        ]*/

@Value
@Wither
public class CashbackOption {
    public UUID id;
    public String name;
    public String title;
    public String description;
    public Boolean selected;
    public Boolean charity;
    public String imageUrl;
    public String selectedUrl;
    public String signature;
    public String paragraph;
}

File: member-service/src/main/java/com/hedvig/memberservice/web/dto/CashbackOption.kt
package com.hedvig.memberservice.web.dto

import java.util.UUID

data class CashbackOption(
    val id: UUID? = null,
    @JvmField
    val name: String? = null,
    val title: String? = null,
    val description: String? = null,
    val selected: Boolean? = null,
    val charity: Boolean? = null,
    val imageUrl: String? = null,
    @JvmField
    val selectedUrl: String? = null,
    @JvmField
    val signature: String? = null,
    @JvmField
    val paragraph: String? = null
)
